internal/compiler/langtype.rs

```diff
@@ -481,23 +481,43 @@ impl ElementType {
         }
     }
 
-    pub fn lookup_type_for_child_element(
+    pub fn can_have_child(&self, name: &str, tr: &TypeRegister) -> bool {
+        match self {
+            Self::Component(component) if component.child_insertion_point.borrow().is_none() => {
+                let base_type = component.root_element.borrow().base_type.clone();
+                if base_type == tr.empty_type() {
+                    false
+                } else {
+                    base_type.can_have_child(name, tr)
+                }
+            }
+            Self::Builtin(builtin) => {
+                if builtin.additional_accepted_child_types.contains_key(name) {
+                    true
+                } else {
+                    !builtin.disallow_global_types_as_child_elements
+                }
+            }
+            _ => true,
+        }
+    }
+
+    pub fn accepts_child_element(
```

Comment:
A documentation comment explaining the return value (when None, when something) would be welcome